syntax = "proto3";

package superhero;

// The gRPC service definition
service SuperHeroService {
  rpc SearchHero(SearchHeroRequest) returns (SearchHeroResponse);
  rpc SubscribeUpdates(SubscribeUpdatesRequest) returns (stream UpdateNotification); // Changed to return a stream
}

// Request message containing the hero name
message SearchHeroRequest {
  string access_token = 1;
  string name = 2;
}

// Response message containing the hero details
message SearchHeroResponse {
  repeated Hero heroes = 1;
}

// Request message for subscribing to updates
message SubscribeUpdatesRequest {
  string access_token = 1;
}

// Notification message for updates
message UpdateNotification {
  string message = 1;
}

// Information of a single hero
message Hero {
  string id = 1;
  string name = 2;
  string powerstats = 3;
  string biography = 4;
  string appearance = 5;
  string work = 6;
  string connections = 7;
  string image_url = 8;
}
